dcpy/lifecycle/ingest/run.py

```diff
@@ -8,6 +9,53 @@
 TMP_DIR = Path("tmp")
 
 
+def update_freshness_or_patch_existing_dataset(
+    config: Config,
+    staging_dir: Path,
+    latest: bool,
+    allow_patch: bool,
+) -> Config:
+    """
+    This function is called after a dataset has been preprocessed, just before archival
+    It's called in the case that the version of the dataset in the config (either provided or calculated)
+      already exists
+
+    The last archived dataset with the same version is pulled in by pandas and compared to what was just processed
+    If they are identical, the last archived dataset has its config updated to reflect that it was checked but not re-archived
+    If they differ, the version is "patched" and a new patched version is archived
+    """
+    new = pd.read_parquet(staging_dir / config.filename)
+    comparison = recipes.read_df(config.dataset)
+    if new.equals(comparison):
+        original_archival_timestamp = recipes.update_freshness(
+            config.dataset_key, config.archival_timestamp
+        )
+        config.archival_timestamp = original_archival_timestamp
+        if latest:
+            recipes.set_latest(config.dataset_key, config.acl)
+        return config
+    else:
+        if not allow_patch:
+            raise FileExistsError(
+                f"Archived dataset '{config.dataset_key}' already exists and has different data. "
+                "Rerun with `allow_patch`=True to patch this version, or specify version manually."
+            )
+        if "." in config.dataset.version:
+            raise ValueError(
+                f"Archived dataset '{config.dataset_key}' already exists and has different data. "
+                "Due to period in version name, cannot patch automatically. "
+                "Specify version manually."
+            )
+        patch = 0
+        patched = config.dataset
+        while recipes.exists(patched):
```

Comment:
seems like a pretty sensible and reusable function to add to recipes?

Reply:
This part is actually just scrapped for now in the next commit, sorry

Reply:
But I agree - if we end up going this route, this should be a recipe function called `patch` or something like that